### ISSUE TITLE
Implement verifyWalk for heartbeat challenge game

### ISSUE THREAD

**Issue by troggy** [leapdao/leap-contracts#279]
# Bounty

We've designed a validator heartbeat challenge game as defined [here](https://docs.leapdao.org/validators/operations/#publishing-heartbeats) ([alternative link](https://github.com/leapdao/docs/pull/61)). Some parts of it are not yet implemented. Including a period walk proof verification. This bounty is all about that.

## Scope
- implement a `PoaOperator.verifyWalk` to verify the period chain: they should be consecutive referencing one another

## Deliverables
- updated code
- updated tests

## Gain for the project
ability to use `minimumPulse` > 0 for validator heartbeat challenges

## Roles
bounty gardener: @troggy / 50 DAI
bounty worker: name / share
bounty reviewer: name / share

## Funded by

Dev Circle

**Comment by troggy** [comment]
@TheReturnOfJan I don't really get the structure of `walkProof`. Can you explain? In a period proof we now have a prev period hash, so given a period proof we can check that the previous period was submitted. Should we provide a proof for that previous as well? Otherwise I don't see how can we check that they are indeed consecutive

**Comment by TheReturnOfJan** [comment]
> So the walkProof is an array like [evenMorePreviousPeriodProof, previousPeriodProof, periodProof]

Yes. You can think of it as array of arrays of bytes32 [[bytes32, bytes32, bytes32], [bytes32, bytes32, bytes32], [bytes32, bytes32, bytes32],...]. The length of the outer array is the length of the walk (how many periods back you go),  and the length of the inner array is the size of the prevPeriod inclusion proof, which is constant (but I do not know of the top of my head what it is). 
The input parameter type is currently bytes, since at the time I did not want to commit to details yet. But maybe it would make sense to check the type of walkProof to bytes32[][].

**Comment by troggy** [comment]
expired. Needs to be resubmitted by gardener
